core/src/main/java/hudson/model/Descriptor.java

```diff
@@ -806,8 +808,31 @@ public boolean configure( StaplerRequest req ) throws FormException {
      *      to keep the client in the same config page.
      */
     public boolean configure( StaplerRequest req, JSONObject json ) throws FormException {
-        // compatibility
-        return configure(req);
+
+        if (Util.isOverridden(Descriptor.class, getClass(), "configure", StaplerRequest.class)) {
+            // compatibility
+            return configure(req);
+        }
+
+        try (BulkChange bc = new BulkChange(this)) {
```

Comment:
IMO it should check `if (BulkChange.contains(this))` and invoke new BulkChange only when needed

Reply:
AFAIK BulkChange by design fully support being nested, doesn't it ?
BulkChange.contains is mostly used by `save` to bypass actual serialization on disk

Reply:
I can't find any other sample of using BulkChange with such an additional contains check

Reply:
I should review it more carefully

Reply:
I also can't imagine any significant use-case to invoke `configure` within an existing BulkChange on the exact same object that would me this `contains` check relevant.